pkg/scheduler/objects/application.go

```diff
@@ -1050,26 +1049,54 @@ func (sa *Application) tryReservedAllocate(headRoom *resources.Resource, nodeIte
 			alloc := newReservedAllocation(Unreserved, reserve.nodeID, unreserveAsk)
 			return alloc
 		}
-		// check if this fits in the queue's head room
-		if !headRoom.FitInMaxUndef(ask.AllocatedResource) {
-			continue
-		}
-		// check allocation possibility
-		alloc := sa.tryNode(reserve.node, ask)
-		// allocation worked fix the result and return
-		if alloc != nil {
-			alloc.Result = AllocatedReserved
-			return alloc
+
+		// Is it daemon set?
+		if ask.GetRequiredNode() == "" {
+			// check if this fits in the queue's head room
+			if !headRoom.FitInMaxUndef(ask.AllocatedResource) {
+				continue
+			}
+
+			// check allocation possibility
+			alloc := sa.tryNode(reserve.node, ask)
+
+			// allocation worked fix the result and return
+			if alloc != nil {
+				alloc.Result = AllocatedReserved
+				return alloc
+			}
+		} else if reserve.node.GetCapacity().FitInMaxUndef(ask.GetAllocatedResource()) {
+			log.Logger().Info("Triggering preemption process for daemon set ask",
+				zap.String("ds allocation key", ask.AllocationKey))
+
+			// try preemption and see if we can free up resource
+			preemptor := NewSimplePreemptor(reserve.node)
+
+			// Are there any victims/asks to preempt?
+			victims := preemptor.GetVictims(ask)
```

Comment:
We only need the `AllocatedResource` size of the ask, we work on just one repeat always.